Tools/i18n/pygettext.py

```diff
@@ -343,6 +344,34 @@ def __waiting(self, ttype, tstring, lineno):
                 return
         if ttype == tokenize.NAME and tstring in opts.keywords:
             self.__state = self.__keywordseen
+            return
+        if ttype == tokenize.STRING:
+            maybe_fstring = ast.parse(tstring, mode='eval').body
+            if not isinstance(maybe_fstring, ast.JoinedStr):
+                return
+            for value in filter(lambda node: isinstance(node, ast.FormattedValue),
+                                maybe_fstring.values):
+                for call in filter(lambda node: isinstance(node, ast.Call),
+                                   ast.walk(value)):
+                    func = call.func
+                    if isinstance(func, ast.Name):
+                        func_name = func.id
+                    elif isinstance(func, ast.Attribute):
+                        func_name = func.attr
+                    else:
+                        continue
+
+                    if func_name not in opts.keywords:
+                        continue
+                    if len(call.args) != 1 or call.keywords:
+                        # what warning should I print when this happens?
```

Comment:
I would assume you can skip processing these entries, IMHO no need to print some sort of warning.

Reply:
I don't really agree with that - I think it's important for these to print warnings as it can allow people to notice incorrect usage of `_()` or even build automated checks around this (although I know that building automations around it comes with its own issues like no guarantee of stable output) so that it can be automatically detected by CI.
Personally I see people mistakenly writing one of these:
```py
_(f"some text with {field}")
_("some text with {field}".format(field="field"))
```
in PRs I review somewhat often and it can be easy to miss. 
I know there are other ways of detecting such usage, some 3rd-party linters perhaps, but considering it just requires coming up with warning message for two prints, I think it should be done.

I'll try adding something tomorrow if that's okay

Reply:
Is it possible to use `safe_eval` for f-string values (just like normal strings)?

Reply:
I added some warnings, I'm not entirely happy that I'm putting a whole f-string in the error but it's probably better than using `ast.unparse()` which results in printing code that isn't formatted the same as in the original file.

> Is it possible to use safe_eval for f-string values (just like normal strings)?

I don't think it can be done that easily as this operates on an AST node generated from a parsed source. The only way would probably be using `ast.unparse()` and then operating on that again, which seems to me like unnecessary trouble when we already have all the needed information in the AST.
If I were to use tokenizer on the unparsed f-string value (as I think that would probably be the only sensible way? I'm not too sure on that though), I would also be operating on the version with formatting different from the original file so tokenizer's warning would contain

Reply:
I have no actual preference here, but I presume you could also use `ast.get_source_segment` to get the original from from the source string (tsring) you have. Anyway, I'm away from the computer, but in general looks good to me.

Reply:
> I presume you could also use ast.get_source_segment to get the original from from the source string (tsring) you have.

I didn't know that function exists, this seems to do exactly what I wanted and I just updated the PR with it, thanks :)